tests/rptest/tests/wasm_identity_test.py

```diff
@@ -28,26 +26,67 @@ def __init__(self, test_context, num_records=1024, record_size=1024):
         self._record_size = record_size
         assert len(self.topics) >= 1
 
-    def wasm_test_plan(self):
-        input_topic = self.topics[0].name
-        mapped_topic = "myoutputtopic"
-        output_topic = construct_materialized_topic(input_topic, mapped_topic)
+    def input_topics(self):
+        """
+        Default behavior is for all scripts to have all topics as input topics
+        """
+        return [x.name for x in self.topics]
+
+    def wasm_test_outputs(self):
+        """
+        The materialized log:
+        [
+          itopic.$script_a_output$,
+        ]
+        Should exist by tests end and be identical to its respective input log
+        """
+        itopic = self.topics[0].name
+        return [[(itopic, "script_a_output")]]
 
-        # The identity transform produces 1 identital record onto a topic for
-        # each input record. The result should be a 1 to 1 mapping between a
-        # source and destination topic, they should be identical when compared
-        basic_script = WasmScript(
-            inputs=[(input_topic, (self._num_records, self._record_size))],
-            outputs=[(output_topic, self._num_records)],
-            script=WasmTemplateRepository.IDENTITY_TRANSFORM)
+    def wasm_xfactor(self):
+        """
+        Multiply with self._num_records to obtain expected record count
+        """
+        return 1
```

Comment:
Is this a TODO?

Reply:
no 1 is just a default value, subclasses can override and provide their own value which one actually does. The method is invoked in the WasmIdentityTest base class

Reply:
Ah. I missed the whole Base-class context. Thanks for clearing it out!